<!--
 The ROS package for CeleX^TM CeleX5-MIPI Dynamic Vision Sensor.

 Copyright (C) 2020  Kehan.Xue<kehan.xue@gmail.com>

 This program is free software: you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation, either version 3 of the License, or
 (at your option) any later version.

 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
-->

<launch>
    <arg name="required" default="false"/>

    <arg name="tf_prefix" default=""/>
    <arg name="frame_id" default="celex5_mipi"/>


    <!--           CeleX5-MIPI Mode             -->
    <!--  Event_Off_Pixel_Timestamp_Mode    0   -->
    <!--  Event_In_Pixel_Timestamp_Mode     1   -->
    <!--  Event_Intensity_Mode              2   -->
    <!--  Full_Picture_Mode                 3   -->
    <!--  Optical_Flow_Mode                 4   -->
    <!--  Optical_Flow_FPN_Mode             5   -->
    <!--  Multi_Read_Optical_Flow_Mode      6   -->

    <arg name="sensor_cfg_file_dir" default="$(find celex5_ros)/config/"/>
    <arg name="fpn_file_dir" default="$(find celex5_ros)/config/"/>

    <arg name="fixed_mode" default="0"/>
    <arg name="is_loop_mode_enabled" default="false"/>
    <arg name="loop_mode1" default="3"/>
    <arg name="loop_mode2" default="0"/>
    <arg name="loop_mode3" default="4"/>

    <arg name="event_frame_time" default="30000"/>
    <arg name="optical_flow_frame_time" default="20"/>
    <arg name="threshold" default="185"/>
    <arg name="brightness" default="150"/>
    <!--    <arg name="contrast" default="1"/>-->
    <arg name="clock_rate" default="100"/>

    <arg name="event_duration_in_loop" default="100"/>
    <arg name="picture_number_in_loop" default="30"/>

    <arg name="event_FPN_file_path" default="$(find celex5_ros)/config/FPN_Optical.txt"/>
    <arg name="frame_FPN_file_path" default="$(find celex5_ros)/config/FPN_2.txt"/>

    <node pkg="celex5_ros" type="celex5_ros_node" name="celex5_mipi" required="true" output="screen">
        <param name="sensor_cfg_file_dir" type="str" value="$(arg sensor_cfg_file_dir)"/>
        <param name="fpn_file_dir" type="str" value="$(arg fpn_file_dir)"/>
        <param name="frame_id" type="str" value="$(arg frame_id)"/>
        <param name="fixed_mode" type="int" value="$(arg fixed_mode)"/>
        <param name="loop_mode1" type="int" value="$(arg loop_mode1)"/>
        <param name="loop_mode2" type="int" value="$(arg loop_mode2)"/>
        <param name="loop_mode3" type="int" value="$(arg loop_mode3)"/>
        <param name="event_frame_time" type="int" value="$(arg event_frame_time)"/>
        <param name="optical_flow_frame_time" type="int" value="$(arg optical_flow_frame_time)"/>
        <param name="threshold" type="int" value="$(arg threshold)"/>
        <param name="brightness" type="int" value="$(arg brightness)"/>
        <param name="clock_rate" type="int" value="$(arg clock_rate)"/>
        <param name="is_loop_mode_enabled" type="bool" value="$(arg is_loop_mode_enabled)"/>
        <param name="event_duration_in_loop" type="int" value="$(arg event_duration_in_loop)"/>
        <param name="picture_number_in_loop" type="int" value="$(arg picture_number_in_loop)"/>
        <param name="event_FPN_file_path" type="str" value="$(arg event_FPN_file_path)"/>
        <param name="frame_FPN_file_path" type="str" value="$(arg frame_FPN_file_path)"/>
    </node>

</launch>
